### DISCUSSION TITLE
x-trap.noscroll adds padding to root element

### DISCUSSION THREAD

**Discussion by LaurinStapf** [alpinejs/alpine#4365] [7. Bugs (archived)] [answered]
Hi, I am currently working on a modal with alpine.js I have noticed that whenever I open the modal, x-trap.noscroll adds a padding to my html root tag when I close the modal it is removed again

What is the reason for this and how can I fix it? 
It worked fine for a while, so I'll declare this as a “bug”

**Modal open:**
`<html lang="en" class="sf-js-enabled" style="overflow: hidden; padding-right: 17px;">`

**Modal closed:**
`<html lang="en" class="sf-js-enabled" style="">`

Would be cool if someone could help me with this problem.

**Comment by SimoTod** [comment]
It has always been that way, as far as I remember but only if your page has a scrollbar before activating the trap. Because the trap sets overflow hidden, it makes the scrollbar disappear so the padding is to compensate that so your page doesn't "jump" sideways. 
Alpine doesn't implement that, it just uses a third party plugin called focus-trap.

**Reply by LaurinStapf** [reply]
Okay, thanks for the information. Actually the padding is a way to much.🙈 that's why my page is jumping when I open the modal.

**Reply by SimoTod** [reply, marked as answer]
Ok, I was actually mistaken. The padding is calculated by alpine, not focus trap. You can find the code here: https://github.com/alpinejs/alpine/blob/ab743bc5668f018db5d410cad4e48b755696ad9e/packages/focus/src/index.js#L201

In theory, it should be the correct value

**Reply by LaurinStapf** [reply]
Okay, interesting. It doesn't seem to calculate properly for me, the page always jumps a little to the right. 

I could imagine that it is due to my styling. I just measured the scrollbar, it is exactly 17pxl

**Comment by LaurinStapf** [comment]
Just looked at the code:

`<body class="flex flex-col font-roboto overflow-y-scroll" style="margin-bottom: 30px;">`

x-trap.noscroll colides with `overflow-y-scroll` from tailwind.css

**Reply by ekwoka** [reply]
That's not really a collision. It's entirely intended.
